### PR TITLE
KUBESAW-16: Label the token secrets

### PR DESCRIPTION
**Pull request by metlos**
Make sure the secret containing the creds for clients connecting between clusters is labeled as belonging to its ToolchainCluster resource.

This will be used in the future to base the creation of the ToolchainClusters on the existence of such secrets.

Associated PRs:
- toolchain-cicd: https://github.com/codeready-toolchain/toolchain-cicd/pull/109
- api: https://github.com/codeready-toolchain/api/pull/423
- host-operator: https://github.com/codeready-toolchain/host-operator/pull/1030
- toolchain-e2e: https://github.com/codeready-toolchain/toolchain-e2e/pull/970

### CODE REVIEW

File: controllers/toolchaincluster/toolchaincluster_controller.go [hunk 1]
```diff
@@ -16,6 +17,10 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
+// TODO: move this to somewhere more appropriate - maybe somewhere in the api module?
+// We can live with this here for now, because this label is not used for anything yet.
+var ToolchainClusterLabel = toolchainv1alpha1.LabelKeyPrefix + "toolchain-cluster"
```
Comment (MatousJobanek) [inline controllers/toolchaincluster/toolchaincluster_controller.go:22]: Yeah, let's move it to the api repo https://github.com/codeready-toolchain/api/blob/master/api/v1alpha1/toolchaincluster_types.go

**Review by MatousJobanek — COMMENTED** [review]
Looks good, but let's also cover the error case when the update would fail.
Are you going to create also a e2e test for it?

**Review by mfrancisc — COMMENTED** [review]
Nice Job 👍 

one minor question - should we add a test case for "secret not found" ?

**Comment by metlos** [comment]
> one minor question - should we add a test case for "secret not found" ?
good point 👍🏼 it's added.

**Review by mfrancisc — APPROVED** [review]
thanks for addressing my comments!

**Pull request merged**